cgi-bin/dashboard-aii.cgi

```diff
@@ -0,0 +1,335 @@
+    return ;
+  }
+  # We unpack the IP address
+  my @tmp_address = unpack('C4',$all_address[0]);
+  my @result;
+  $result[0]=sprintf ("%02X%02X%02X%02X",$tmp_address[0], $tmp_address[1],
+                                         $tmp_address[2], $tmp_address[3]);
+  $result[1]=sprintf ("%u.%u.%u.%u",$tmp_address[0], $tmp_address[1],
+                                    $tmp_address[2], $tmp_address[3]);
+  return @result;
+}
+
+
+#########################################################################
+sub Initialize {
+#########################################################################
+
+  @profiles=();
+
+  opendir(DIR,$profiles_dir);
+
+  # find all profiles in directory
+  push @profiles,map { s/\.json$//; s/^$profile_prefix//; $_ .=''; } sort(grep(/\.json$/, readdir(DIR)));
+
+  closedir(DIR);
+  # Load the configurations list
+  opendir(DIR, $pxelinux_dir);
+  @cfg = sort(grep(/(\.cfg$)|(default)/, readdir(DIR)));
+  closedir(DIR);
+}
+
+#########################################################################
+sub GetHosts {
+#########################################################################
+
+  my (@all, $k, $json);
+
+  for $k (@profiles) {
+
+    my $hostname = $k;
+
+    my ($hexaddr,$dotaddr) = GetHexAddr($hostname);
+
+    my $link = readlink("$pxelinux_dir/$hexaddr");
+    my $existing_cfg = $link ? $link : "";
+
+    push @all ,{'hostname' => $hostname, 'hexaddr' => $hexaddr, 'dotaddr' => $dotaddr, 'bootcfg' => $existing_cfg };
+
+  }
+
+  my $json =  JSON::XS->new->pretty->encode({hosts => \@all, available_cfg => \@cfg});
+
+  print "Content-type: application/json\n\n$json";
+
+}
+
+#########################################################################
+sub GetProfile {
+#########################################################################
+
+  if ($_[0] =~ /^(.*)$/) {
+    $_[0] = $1;
+    my $hostname = $_[0];
+    my $file = "$profiles_dir/$profile_prefix$hostname.json";
+
+      my $json_text = do {
+        open(my $json_fh, "<:encoding(UTF-8)", $file) or die("Can't open $file: $!\n");
+        local $/;
+        <$json_fh>
+      };
+
+      my $profile = JSON::XS->new->decode($json_text);
+      
+      #Remove sensitive sections
+      $profile->{'software'} = ();
+
+      my $json =  JSON::XS->new->pretty->allow_nonref->encode($profile);
+
+      print "Content-type: application/json\n\n$json";
+    }
+}
+
+#########################################################################
+sub Configure {
+#########################################################################
+
+  $ENV{PATH}="/bin:/usr/bin:/sbin:/usr/bin:/usr/sbin";
+
+  my $result = '';
+  print "Content-type: text/plain\n\n";
+
+  $_[0] =~ /^(configure|install|reinstall|boot)$/;
+  $_[0] = $1;
+
+  $_[1] =~ /^(.*)$/;
+  $_[1] = $1;
+
+  my $action = $_[0];
+  my $hostname = $_[1];
+
+  if ($action eq 'reinstall') {
+    my $ok = system ("/usr/bin/sudo", "/usr/sbin/aii-shellfe", "--remove", $hostname);
```

Comment:
use `CAF::process`, check exit code amd i think you can do `--remove --configure --install` in one go